lib/datadog/tracing/contrib/span_attribute_schema.rb

```diff
@@ -22,6 +22,76 @@ def default_span_attribute_schema?
           Datadog.configuration.tracing.span_attribute_schema ==
             Tracing::Configuration::Ext::SpanAttributeSchema::DEFAULT_VERSION
         end
+
+        # TODO: implement function in all integrations with spankind
+        def set_peer_service(span)
+          set_peer_service?(span) && set_peer_service_from_source(span)
+          # TODO: add logic for remap as long as the above expression is true
+        end
+
+        # set_peer_service?: checks to see if any edited peer.service tags exist so that they are not overwritten
+        def set_peer_service?(span)
+          ps = span.get_tag(Tracing::Metadata::Ext::TAG_PEER_SERVICE)
+          if ps && (ps != '')
+
+            # if peer.service is not equal to span.service we know it is edited
+            if ps != span.service
+              span.set_tag(Tracing::Metadata::Ext::TAG_PEER_SERVICE_SOURCE, Tracing::Metadata::Ext::TAG_PEER_SERVICE)
+              return false
+            end
+
+            # if span.service is not the global service, we know it was changed to change the peer.service value
+            if (ps == span.service) && (span.service != Datadog.configuration.service)
```

Comment:
I believe `ps == span.service` is not needed here, given we return early in the `if` condition prior if `ps != span.service`:
```suggestion
            if span.service != Datadog.configuration.service
```

Reply:
That is correct, fixed!